API/StoryAPI.cs

```diff
@@ -166,6 +166,30 @@ public static void Map(WebApplication app)
                 }
                 return Results.Ok(searchResults);
             });
+
+            //GET STORIES BY CATEGORY
+            app.MapGet("/stories/categories/{categoryId}", (FanFusionDbContext db, int categoryId) =>
+            {
+                var category = db.Categories.FirstOrDefault(c => c.Id == categoryId);
+
+                // Check if the category exists
+                if (category == null)
+                {
+                    return Results.NotFound($"Category with ID {categoryId} not found.");
+                }
+
+                var stories = db.Stories
+                    .Include(s => s.Category)
+                    .Where(c => c.CategoryId == categoryId)
+                    .ToList();
+
+                if (!stories.Any())
+                {
+                    return Results.NotFound("No stories found for this category.");
+                }
+
```

Comment:
Add this var storyDTOs = stories
        .Select(story => new StoryDTO(story))
        .OrderByDescending(dto => dto.DateCreated)
        .ToList(); and return storyDTOs instead of stories in line 191

Reply:
@yarelismartin done

Reply:
@yarelismartin done